### PR TITLE
Post-release 0.7.0

### PR DESCRIPTION
**Pull request by sumerman**
## Description

- Post-release SQL bits.
- Switching to the post-release dev version.
- Resolves https://github.com/timescale/promscale_extension/issues/527

## Merge requirements

Waiting for `timescaledev/promscale-extension:0.7.0-ts2.8.1-pg` images

Please take into account the following non-code changes that you may need to make with your PR:

- [x] CHANGELOG entry for user-facing changes
- [x] Updated the relevant documentation

### CODE REVIEW

**Comment by sumerman** [comment]
In the interest of unblocking further merges in this repo, I suppose it'd be acceptable to merge even before we have `timescaledev/promscale-extension:0.7.0-ts2.8.1-pg` images ready. We can update the test config when they are.

File: INSTALL.md [hunk 1]
```diff
@@ -104,7 +104,7 @@ For bare-metal installations, the full instructions for setting up PostgreSQL, T
     sudo apt-get install -y git
     git clone https://github.com/timescale/promscale_extension
     cd promscale_extension
-    git checkout 0.7.0
+    git checkout 0.7.1-dev
```
Comment (cevian) [inline INSTALL.md:107]: why are we changing the instructions to install a non-release? Also, I don't think thats an actual tag.

File: Cargo.toml [hunk 2]
```diff
@@ -22,7 +22,7 @@ panic = "unwind"
 # If your tests fail with SIGSEGV try and use 1.64 or newer.
 # Sadly, we can't just drop it because it's a workaround for
 # another issue https://github.com/tcdi/pgx/pull/208
-lto = "thin"
+#lto = "thin"
```
Comment (cevian) [inline Cargo.toml:25]: why are we turning lto off?

**Review by cevian — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 2 above)
Comment (sumerman) [reply Cargo.toml:25]: I keep accidentally committing it. Thanks for catching it!

File: INSTALL.md (hunk 1 above)
Comment (sumerman) [reply INSTALL.md:107]: You are correct. `./update-version.sh` does it, but it needs to leave `INSTALL.md` alone on the post-release run. I'll fix it.

**Review by cevian — APPROVED** [review]

**Pull request merged**